go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -66,10 +84,95 @@ func TestComputeDirectory(t *testing.T) {
 	}
 }
 
-func createFile(t *testing.T, executable bool, contents string) (string, error) {
+type testSymlinkCreationResult struct {
+	symlink string
+	target  string
+}
+
+func (sc *testSymlinkCreationResult) cleanup() {
+	os.RemoveAll(sc.symlink)
+	os.RemoveAll(sc.target)
+}
+
+func TestComputeSymlinks(t *testing.T) {
+	tests := []struct {
+		name   string
+		target interface{} // If unspecified, this is an invalid symlink
```

Comment:
Some cleaner options for doing this:
1. What we usually do is make the test struct a union of the test parameters. There is nothing wrong with having an IsDirectory option, or an IsExecutable option that only applies to files and not directories.
2. If you really want to have a single field of variable type, the golang pattern is to create a common interface (say, CreateSymlink) and have your types implement it.
I slightly prefer option 2 in this case, because I suspect that if your createSymlink is split into two or three functions (you can create a different type for a dangling symlink), it will look much simpler.

Reply:
Ah, actually i split this into three tests now. Sorry for mixing the logic before, hopefully it now gets simpler?

Reply:
I think flattening things out will be simplest. Also, I don't think the rest of the package can use the test code, at least not if compiled with Bazel (not sure how go build works), so no need to prefix with test.

Reply:
Ack, removed `testFileParams`